mathics/core/expression.py

```diff
@@ -729,43 +729,45 @@ def get_sort_key(self, pattern_sort=False) -> list:
                 else:
                     pattern += 20
             if pattern > 0:
-                return [
-                    2,
-                    pattern,
-                    1,
-                    1,
-                    0,
-                    head.get_sort_key(True),
-                    tuple(element.get_sort_key(True) for element in self._elements),
-                    1,
-                ]
+                return tuple(
+                    (
+                        2,
+                        pattern,
+                        1,
+                        1,
+                        0,
+                        head.get_sort_key(True),
+                        tuple(element.get_sort_key(True) for element in self._elements),
+                        1,
+                    )
+                )
 
             if head is SymbolPatternTest:
                 if len(self._elements) != 2:
-                    return [3, 0, 0, 0, 0, head, self._elements, 1]
-                sub = self._elements[0].get_sort_key(True)
+                    return tuple((3, 0, 0, 0, 0, head, self._elements, 1))
+                sub = list(self._elements[0].get_sort_key(True))
```

Comment:
This is probably the reason for using `list` instead of `tuple`: Here the return value is obtained from calling the method on one of the elements, and then modifying it. Probably there is also a better way to implement the same...

Reply:
This kind of behavior where you want the 0th element sort key and modify the portion of based on `head` should be wrapped inside a function.

Reply:
@rocky, I am not sure how to implement what you suggested. In any case, in a following round, I would like to reimplement all this chuck of code, by specializing the class for Patterns.